tools/system_libs.py

```diff
@@ -441,20 +441,20 @@ def require_dlmalloc(what):
         shared.exit_with_error('only dlmalloc is possible when using %s' % what)
 
     extra = ''
+    if shared.Settings.DEBUG_LEVEL >= 3:
+      extra += '_debug'
+    if not shared.Settings.SUPPORT_ERRNO:
+      # emmalloc does not use errno anyhow
+      if base != 'emmalloc':
+        extra += '_noerrno'
     if shared.Settings.USE_PTHREADS:
       extra += '_threadsafe'
-      require_dlmalloc('pthreads')
     if shared.Settings.EMSCRIPTEN_TRACING:
       extra += '_tracing'
       require_dlmalloc('tracing')
-    if shared.Settings.DEBUG_LEVEL >= 3:
-      extra += '_debug'
+      require_dlmalloc('pthreads')
```

Comment:
Shouldn't this line be after 451?

If so I guess this shows we are not testing this.

Reply:
Oops, yeah, thanks.

And yeah, we don't test emmalloc errors on these corner cases I guess. emmalloc is not a default option so it's gotten less attention I guess.